### ISSUE TITLE
vm_compute/native_compute does not properly compute with int63 in functors

### ISSUE THREAD

**Issue by fajb** [coq/coq#12519]
Hello,
In the following code, `vm_compute` does not normalise the term and `native_compute` fails.
The problem seem to arise when using native integers through functors.

```coq
Module Type K.
  Axiom t: Type.
  Axiom land  : t -> t -> t.
End K.

Module M (K:K).

      Definition mask (k m: K.t) := K.land k m.

End M.

Require Int63.
Module KInt.
  Definition t := Int63.int.
  Definition land:= Int63.land.
End KInt.

Module MInt := M(KInt).

Import Int63.
Eval vm_compute in ((MInt.mask 1 1 == 1)%int63).
(**      = (1 land 1 == 1)%int63     : bool *)

Eval native_compute in ((MInt.mask 1 1 == 1)%int63).
(** Unbound value const_Nbug_KInt_land *)
```

#### Coq Version
8.12+alpha

**Comment by JasonGross** [comment]
This can be turned into a failure-at-Qed-time, btw:
```coq
Module Type K.
  Axiom t: Type.
  Axiom land  : t -> t -> t.
End K.

Module M (K:K).

      Definition mask (k m: K.t) := K.land k m.

End M.

Require Int63.
Module KInt.
  Definition t := Int63.int.
  Definition land:= Int63.land.
End KInt.

Module MInt := M(KInt).

Import Int63.
Lemma foo : ((MInt.mask 1 1 == 1)%int63) = true.
Proof.
  vm_compute.
  exact (eq_refl true).
Qed. (* Error: The term "eq_refl" has type "true = true" while it is expected to have type "(MInt.mask 1 1 =? 1)%int63 = true". *)
```

**Comment by SkySkimmer** [comment]
@maximedenes you're assigned to this issue, does that mean you're working on it?

**Comment by maximedenes** [comment]
> you're assigned to this issue, does that mean you're working on it?

Not at the moment, though I will when I find some time. Meanwhile, I'll unassign, in case anybody also would like to look at it.

**Comment by JasonGross** [comment]
This has been fixed as part of #17663 / #17686, right?
